rewriter/flatten.cc

```diff
@@ -313,23 +323,40 @@ class FlattenWalk {
         }
         classDef->rhs = addClassDefMethods(ctx, std::move(classDef->rhs), classDef->loc);
         return classDef;
-    };
+    }
+
+    bool shouldMoveSend(unique_ptr<ast::Send> &send) {
```

Comment:
This seems super bespoke to me. Do I understand it right that meands we _don't_ actually  introduce an invariant?

Reply:
`shouldMoveSend` only applies to the set of things that we remove in the namer
(specifically [in this `postTransformSend`](https://github.com/sorbet/sorbet/blob/master/namer/namer.cc#L454)). If it did not, then we would just need to manage symbol visibility in a slightly different way.

Reply:
This seems to be a big crosstalk & action at a distance.
Let's either remove it(i'd prefer that) or document it & make both namer & resolver use the same helper method to detect it.